server/routes/statistics.js

```diff
@@ -0,0 +1,226 @@
+const express = require("express");
+const User = require('../models/user');
+const Match = require('../models/match');
+const auth = require("../middleware/auth");
+const NodeCache = require('node-cache');
+const router = express.Router();
+const requestCache = new NodeCache({stdTTL: 30, checkperiod: 5 });
+const users = new Map();
+const sortBy = new Map();
+
+async function requestAndCache() {
+	
+	console.log("Fetching new data");
+	await User.find({}, function (err, allUsers) {

```

Comment:
Instead of getting all users - have you thought about using Mongodb queries (sortBy, etc.) and then caching those specific queries, so that you don't need to use a lot of space to constantly store all the users?

Reply:
If I was using mongdb queries to sort users, then I would have to call the db more often. I don't see how it would take up less space to store the queries instead of the sorted lists. I guess this is caching an extra list (the unsorted version of the users) but considering the the users can be sorted in a dozen ways, I don't think 13*numUsers is significantly different from 12*numUsers. On the other hand, getting the users all at once and then sorting it for each separate query decreases the io needed to/from the db.  This also ensures that each user sees data that is consistent (ie, each sort is using the same data as opposed to one using outdated data and another using up to date data). The only improvement that might be worth considering is making the call to the db async and using old data to service requests until the promise.then updates the data. This will ensure that users don't have to ever wait for the db call to finish. But I am not sure if nodejs continues processing requests while a promise from a previous request is still running (ie, Will nodejs continue processing the next request if a promise that began in the current request is still running after the current request has finished executing?).